iPetCare/API/Controllers/ExaminationsController.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Application.Interfaces;
+using Application.Dtos.Examinations;
+using Application.Services.Utilities;
+using Microsoft.AspNetCore.Authorization;
+using Domain.Models;
+using API.Security;
+
+namespace API.Controllers
+{
+    [ApiController]
+    public class ExaminationsController : BaseController
+    {
+        private readonly IExaminationService _examinationsService;
+        public ExaminationsController(IExaminationService examinationsService)
+        {
+            _examinationsService = examinationsService;
+        }
+
+        [Produces(typeof(ServiceResponse<ExaminationsCreateExaminationDtoResponse>))]
+        [AuthorizeRoles(Role.Administrator, Role.Vet, Role.Owner)]
+        [HttpPost]
+        public async Task<IActionResult> CreateExamination(ExaminationsCreateExaminationDtoRequest dto)
+        {
+            var response = await _examinationsService.CreateExaminationAsync(dto);
+            return SendResponse(response);
+        }
+
+        [Produces(typeof(ServiceResponse<ExaminationsGetAllExaminationsDtoResponse>))]
+        [AuthorizeRoles(Role.Administrator)]
+        [HttpGet]
+        public async Task<IActionResult> GetAllExaminations()
+        {
+            var response = await _examinationsService.GetAllExaminationsAsync();
+            return SendResponse(response);
+        }
+        [Produces(typeof(ServiceResponse<ExaminationsGetAllExaminationsDtoResponse>))]
+        [AuthorizeRoles(Role.Administrator, Role.Vet, Role.Owner)]
+        [HttpGet("{petId}")]
+        public async Task<IActionResult> GetPetAllExaminations(string petId)
+        {
+            var response = await _examinationsService.GetPetAllExaminationsAsync(petId);
+            return SendResponse(response);
+        }
+        [Produces(typeof(ServiceResponse<ExaminationsGetAllExaminationsDtoResponse>))]
+        [AuthorizeRoles(Role.Administrator, Role.Vet, Role.Owner)]
+        [HttpGet("{petId}/{examinationId}")]
+        public async Task<IActionResult> GetExamination(string petId, string examinationId)
```

Comment:
tutaj powinny być Guid'y a nie stringi. Analogicznie w każdym dtosie. Jeśli model (tj w tym wypadku "Pet.cs" i "Examination.cs") mają Id jako Guid to kontrolery przyjmują/zwracają Guid a nie stringi :D